nautobot/extras/jobs.py

```diff
@@ -73,17 +76,17 @@
 logger = logging.getLogger(__name__)
 
 
-class BaseJob:
-    """Base model for jobs (reports, scripts).
+class RunJobTaskFailed(Exception):
+    """Celery task failed for some reason."""
+
+
+class BaseJob(Task):
+    """Base model for jobs.
 
     Users can subclass this directly if they want to provide their own base class for implementing multiple jobs
     with shared functionality; if no such sharing is required, use Job class instead.
```

Comment:
If we don't have an issue tracking this, we should IMO - let's get rid of the `BaseJob`/`Job` separation and instead support a `Meta.abstract` property that any Job can define.

Reply:
Tracked as follow-up item